bindings/cpp/logger.cpp

```diff
@@ -156,32 +154,35 @@ void dnet_node_set_trace_id(dnet_logger *logger, uint64_t trace_id, int tracebit
 			attributes.insert(std::make_pair(std::string("backend_id"), blackhole::log::attribute_t(backend_id)));
 		}
 
-		new (local_attributes) scoped_attributes_t(*logger, std::move(attributes));
+		scoped_attributes[scoped_count] = new scoped_attributes_t(*logger, std::move(attributes));
 
 		// Set all bits to ensure that it has tracebit set
 		backend_trace_id_hook = scoped_trace_id_hook[scoped_count];
-	} catch (...) {
-		local_attributes = NULL;
+		++scoped_count;
+	} catch (const std::exception &e) {
+		dnet_log_only_log(logger, DNET_LOG_ERROR,
+			"%s: trace_id: %08llx, tracebit: %d, backend_id: %d, caught exception: %s",
+			__func__, (unsigned long long)trace_id, tracebit, backend_id, e.what());
 	}
-
-	++scoped_count;
 }
 
 void dnet_node_unset_trace_id()
 {
 	using namespace blackhole_scoped_attributes;
 
-	--scoped_count;
+	if (scoped_count > 0) {
```

Comment:
This forces changing all instances of `dnet_node_unset_trace_id()`, I had that commit, but removed, it is really useless.

I'm not totally against it, but let's separate it into its own pull request, this one unfortunately already contains enough unrelated changes.